packages/devtools_app/lib/src/memory/memory_events_pane.dart

```diff
@@ -0,0 +1,255 @@
+  @override
+  MemoryEventsPaneState createState() => MemoryEventsPaneState();
+}
+
+class MemoryEventsPaneState extends State<MemoryEventsPane>
+    with AutoDisposeMixin, OnChartValueSelectedListener {
+  ScatterChartController _controller;
+
+  MemoryController _memoryController;
+
+  MemoryTimeline get _memoryTimeline => _memoryController.memoryTimeline;
+
+  ColorScheme colorScheme;
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+
+    _memoryController = Provider.of<MemoryController>(context);
+    // TODO(jacobr): this is an ugly way to be using the theme. It would be
+    // better if the controllers weren't involved with the color scheme.
+    colorScheme = Theme.of(context).colorScheme;
+
+    _initController(colorScheme);
+
+    cancel();
+
+    addAutoDisposeListener(_memoryTimeline.sampleAddedNotifier, () {
+      _setupEventsChartData(colorScheme);
+      _processAndUpdate();
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if (_controller != null) {
+      colorScheme = Theme.of(context).colorScheme;
+      _setupEventsChartData(colorScheme);
+      final hasData = _controller.data.dataSets.first.getEntryCount() > 0;
+      if (hasData) {
+        return Container(child: ScatterChart(_controller), height: 50);
+      }
+    }
+
+    return const SizedBox(width: denseSpacing);
+  }
+
+  void _initController(ColorScheme colorScheme) {
+    final desc = Description()..enabled = false;
+    _controller ??= ScatterChartController(
+      axisLeftSettingFunction: (axisLeft, controller) {
+        axisLeft
+          ..position = YAxisLabelPosition.OUTSIDE_CHART
+          ..drawGridLines = false
+          ..granularityEnabled = true
+          // Set to baseline min and auto track max axis range.
+          ..setStartAtZero(true);
+      },
+      axisRightSettingFunction: (axisRight, controller) {
+        axisRight.enabled = false;
+      },
+      legendSettingFunction: (legend, controller) {
+        legend.enabled = false;
+      },
+      xAxisSettingFunction: (xAxis, controller) {
+        xAxis
+          ..enabled = true
+          ..position = XAxisPosition.TOP
+          // Hide the text (same as background).
+          ..textColor = colorScheme.defaultBackgroundColor
+          ..drawAxisLine = true
+          ..drawGridLines = true;
+      },
+      backgroundColor: colorScheme.defaultBackgroundColor,
+      maxVisibleCount: 1000,
+      maxHighlightDistance: 50,
+      selectionListener: this,
+      description: desc,
+      minOffset: 0,
+    );
+
+    _controller.setViewPortOffsets(
+        defaultSpacing * 3, denseSpacing, defaultSpacing, 0);
+  }
+
+  /// Override onNothingSelected and onValueSelected for selectionListener.
+  @override
+  void onNothingSelected() {
+    // TODO: implement onNothingSelected
+    print('>>>> onNothingSelected');
```

Comment:
resolve these TODOs and remove the prints.

Reply:
onNothingSelected, etc. are derived from an abstract class will implement soon when next PR with legend is ready.